src/util_t3.h

```diff
@@ -0,0 +1,13 @@
+#ifndef __UTIL_T3_H__
```

Comment:
Maybe just call this `util.h`? Could be a good place to put utility functions (that don't belong in `contrib`) in the future.

Reply:
I've moved it to the already existing `include/utils.h` header and added `src/utils.cu` for the kernels.

Reply:
Thanks.